### ISSUE TITLE
Fatal error terminated IDE

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#4006]
Using a recent Wrench build of master from today: 

Visual Studio 2017 for Mac (Preview) 7.5 Preview (7.5 build 654)

I deleted a file from Main.sln, and then was editing a C# file and saved the changes. A fatal error occured and the IDE closed.

```
FATAL ERROR [2018-03-01 12:36:09Z]: An unhandled exception has occured. Terminating Visual Studio? True
System.OperationCanceledException: The operation was canceled.
  at System.Threading.CancellationToken.ThrowOperationCanceledException () [0x00000] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/CancellationToken.cs:482 
  at System.Threading.CancellationToken.ThrowIfCancellationRequested () [0x00008] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/CancellationToken.cs:469 
  at MonoDevelop.CSharp.Completion.Provider.ProtocolMemberCompletionProvider.TryDetermineOverridableProtocolMembers (Microsoft.CodeAnalysis.SemanticModel semanticModel, Microsoft.CodeAnalysis.SyntaxToken startToken, Microsoft.CodeAnalysis.Accessibility seenAccessibility, System.Collections.Generic.ISet`1[Microsoft.CodeAnalysis.ISymbol]& overridableMembers, System.Threading.CancellationToken cancellationToken) [0x00078] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/ProtocolMemberCompletionProvider.cs:229 
  at MonoDevelop.CSharp.Completion.Provider.ProtocolMemberCompletionProvider+<CreateCompletionData>d__1.MoveNext () [0x0000d] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/ProtocolMemberCompletionProvider.cs:119 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:152 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_1 (System.Object state) [0x00000] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1034 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x0000d] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1306 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:957 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:904 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1283 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:856 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1211 
[ERROR] FATAL UNHANDLED EXCEPTION: System.OperationCanceledException: The operation was canceled.
  at System.Threading.CancellationToken.ThrowOperationCanceledException () [0x00000] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/CancellationToken.cs:482 
  at System.Threading.CancellationToken.ThrowIfCancellationRequested () [0x00008] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/CancellationToken.cs:469 
  at MonoDevelop.CSharp.Completion.Provider.ProtocolMemberCompletionProvider.TryDetermineOverridableProtocolMembers (Microsoft.CodeAnalysis.SemanticModel semanticModel, Microsoft.CodeAnalysis.SyntaxToken startToken, Microsoft.CodeAnalysis.Accessibility seenAccessibility, System.Collections.Generic.ISet`1[Microsoft.CodeAnalysis.ISymbol]& overridableMembers, System.Threading.CancellationToken cancellationToken) [0x00078] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/ProtocolMemberCompletionProvider.cs:229 
  at MonoDevelop.CSharp.Completion.Provider.ProtocolMemberCompletionProvider+<CreateCompletionData>d__1.MoveNext () [0x0000d] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/ProtocolMemberCompletionProvider.cs:119 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:152 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_1 (System.Object state) [0x00000] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1034 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x0000d] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1306 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:957 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:904 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1283 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:856 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1211 
```

> VS bug #574346

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.5.0.757 (d15-7)

Author: Mike Krüger
Commit: 378e839df8c05a250550efdf566d4024b2d7bc8f (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.6.0.5 (master)

Pull Request #4018 merged by: Lluis Sanchez
Author: mono
Commit: 6236b5749faabbba545e2e77108fdc37646c57d2 (mono/monodevelop)

**Comment by slluis** [comment]
@mkrueger should this also be merged to 15.6?

**Comment by mkrueger** [comment]
I would like to  - builds:

https://code.internalx.com/Wrench/BuildDetails?buildId=e19853ed6aa9971dd823f36c67ea714

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.4.1.23 (d15-6)

Pull Request #4019 merged by: Lluis Sanchez
Author: mono
Commit: f452ea42c129d8109b1cc6f38ddab55a5f63d807 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #4018 merged by: Lluis Sanchez
Author: mono
Commit: 6236b5749faabbba545e2e77108fdc37646c57d2 (xamarin/monodevelop)
